File: Programmers/[42583] 다리를 지나는 트럭/KingDonggyu/P_42583.swift
import Foundation

func solution(_ bridge_length:Int, _ weight:Int, _ truck_weights:[Int]) -> Int {
    
    var answer: Int = 0
    var w: Int = 0
    var q: [Int] = []
    var truck: [Int] = truck_weights
    
    for _ in 0..<bridge_length {
        q.append(0)
    }
    
    while true {
        w -= q.removeFirst()
        if truck.count > 0 && truck[0] + w <= weight {
            var x = truck.removeFirst()
            q.append(x)
            w += x
        } else {
            q.append(0)
        }
        
        answer += 1
        if w == 0 { break }
    }
    
    return answer
}

File: Programmers/[42583] 다리를 지나는 트럭/KingDonggyu/P_42583.py
def solution(bridge_length, weight, truck_weights):
    answer = 0; w = 0
    q = [0 for _ in range(bridge_length)]
    while True:
        w -= q.pop(0)
        if truck_weights and truck_weights[0] + w <= weight:
            x = truck_weights.pop(0)
            q.append(x)
            w += x
        else:
            q.append(0)
            
        answer += 1
        if w == 0: break
    
    return answer